core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -217,12 +218,21 @@ object Battle {
         }
     }
 
-    private fun tryGetCultureFromHonor(civUnit:ICombatant, barbarianUnit:ICombatant){
-        if(barbarianUnit.isDefeated() && barbarianUnit is MapUnitCombatant
-                && barbarianUnit.getCivInfo().isBarbarian()
-                && civUnit.getCivInfo().hasUnique("Gain Culture when you kill a barbarian unit"))
-            civUnit.getCivInfo().policies.addCulture(
-                    max(barbarianUnit.unit.baseUnit.strength, barbarianUnit.unit.baseUnit.rangedStrength))
+    private fun tryGetCultureFromHonor(civUnit:ICombatant, enemyUnit:ICombatant){
+        //Aztecs get half the strength of the unit killed in culture and honor opener does the same thing. 
+        //They stack. So you get culture equal to 100% of the dead unit's strength.
+        if(enemyUnit.isDefeated() && enemyUnit is MapUnitCombatant){
+            val civInfo = civUnit.getCivInfo()
+            val cultureAmount = max(enemyUnit.unit.baseUnit.strength, enemyUnit.unit.baseUnit.rangedStrength)/2
+            if (enemyUnit.getCivInfo().isBarbarian() && civInfo.hasUnique("Gain Culture when you kill a barbarian unit"))
+                civInfo.policies.addCulture(cultureAmount)
+            if (!enemyUnit.getUnitType().isCivilian() && civInfo.hasUnique("Gains culture from each enemy unit killed")) {
+                civUnit.getCivInfo().policies.addCulture(cultureAmount)
+                if (civUnit is MapUnitCombatant)
+                    for (unique in civUnit.unit.baseUnit.uniqueObjects.filter { it.placeholderText == "Heals [] damage if kills an enemy" })
+                        civUnit.unit.healBy(unique.params[0].toInt())
```

Comment:
This is in a different part of the code, no? Why is this under "gain culture"?

Reply:
Thought about unification and forgot to rename. 
Remove and use tryHealAfterAttacking instead? Like this at 78 line
```
        // Add culture when defeating a barbarian when Honor policy is adopted 
        // or any enemy military unit with Sacrificial captives unique (can be either attacker or defender!)
        tryGetCultureFromHonor(attacker, defender)
        tryGetCultureFromHonor(defender, attacker)
        tryHealAfterAttacking(defender, attacker)
        tryHealAfterAttacking(attacker, defender)
```

Reply:
Yup
There are already units that do that IIRC